src/Orleans.Core/Configuration/Legacy/LegacyConfigurationExtensions.cs

```diff
@@ -6,11 +6,15 @@
 using Orleans.Runtime.Configuration;
 using Orleans.Hosting;
 using Orleans.Runtime;
+using Orleans.Providers;
 
 namespace Orleans.Configuration
 {
     internal static class LegacyConfigurationExtensions
     {
+        private const int ClusterClientDefaultProviderInitStage = 1000;
+        private const int ClusterClientDefaultProviderStartStage = 2000;
```

Comment:
avoid use hard coded stage number. Isn't you suppose to have a `ClientLifeCycleStage` class, use it here?

Reply:
Currently, the only thing in the lifecycle is the providers, so these are just default stages to wire them up with.  

We need to think through if and how we want to use lifecycle in the client.  If we do fully utilized the lifecycle to control client startup/shutdown, then we'll need to merge these values with the existing client lifecycle logic.

Reply:
I do want lifecycle on the client. I'm not sure how lifecycle stages should be shared in code. Eg, stream providers should be able to subscribe to the same lifecycle on both client and silo, I would imagine. Maybe there'd be a separate, shared lifecycle which those providers can subscribe to?

Reply:
For clarification these are just placeholders until we've time to think through how to implement a client lifecycle.  Having said that:

Do you mean subscribe to the same lifecycle or same lifecycle stage?

I don't think they can have the same lifecycle as the silo and client are, in theory, entirely separate, and in most cases even different processes.  While I've considered some sort of distributes startup logic for cluster start, that's a distant consideration for possible future enhancements, and even in that model I'm unconvinced clients can reliably take part in the cluster lifecycle unless we assume clients are part of the cluster.

As far as stages, as we flesh out the client lifecycle, I do think it would be good to mirror the stages where applicable, but I'm not yet familiar enough with the client initialization/shutdown logic to be confident of any design decisions without further investigation.  If you've more familiarity and can point out which stages of the silo lifecycle should be part of the client lifecycle can attempt a more accurate set of stages.

Reply:
perhaps they could use a separate lifecycle which is shared, like `IStreamProviderLifecycle` or `IProviderLifecycle` or something more broad, like `ISystemLifecycle`?

Separate client and silo lifecycles might not be needed: it seems fine to me if a particular lifecycle stage isn't mapped to any actual event on the client.

Reply:
Ah, I see.  I'm not 100% opposed to this, but I think we should delay such a decision.  My reasoning is that most provider are silo only, and those that can also run on the client will require different initialization logic when running on client vs silo.  This has been a lingering issue in the stream providers which run on both.  I think having two distinct lifecycles is actually the right thing to do as it supports distinct initialization for the distinct environments.  In any case, I am of the opinion that we'll be in a better position to judge this after we've the experience of porting some of our providers to use lifecycles.